server/aesdsocket.c

```diff
@@ -63,7 +70,7 @@ typedef struct
 	bool timer_thread_success;
 }timer_thread_data;
 
-
+#ifndef USE_AESD_CHAR_DEVICE
```

Comment:
@MuktaDarekar think about how you could put less of these build switches in your code.  Ideally you'd be able to keep all of the functions related to timer compiling, just disable the part which initializes and runs, hopefully just a few lines of code total.